wp-admin/js/widgets/media-image-widget.js

```diff
@@ -80,30 +80,89 @@
 		 * @returns {void}
 		 */
 		selectMedia: function selectMedia( event ) {
-			var control = this, selection, mediaFrame;
+			var control = this, selection, mediaFrame, library, CustomizedDisplaySettingsLibrary, customizedDisplaySettings;
 
 			event.preventDefault();
 
 			selection = new wp.media.model.Selection( [ control.selectedAttachment ] );
 
+			/*
+			 * Copy current display settings from the widget model to serve as basis
+			 * of customized display settings for the current media frame session.
+			 * Changes to display settings will be synced into this model, and
+			 * when a new selection is made, the settings from this will be synced
+			 * into that AttachmentDisplay's model to persist the setting changes.
+			 */
+			customizedDisplaySettings = new Backbone.Model( {
+				align: control.model.get( 'align' ),
+				size: control.model.get( 'size' ),
+				link: control.model.get( 'link_type' ),
+				linkUrl: control.model.get( 'link_url' )
+			} );
+
+			/**
+			 * Library which persists the customized display settings across selections.
+			 *
+			 * @class
+			 */
+			CustomizedDisplaySettingsLibrary = wp.media.controller.Library.extend( {
+
+				/**
+				 * Sync changes to the current display settings back into the current customized
+				 *
+				 * @param {Backbone.Model} displaySettings Modified display settings.
+				 * @returns {void}
+				 */
+				handleDisplaySettingChange: function handleDisplaySettingChange( displaySettings ) {
+					customizedDisplaySettings.set( displaySettings.attributes );
+				},
+
+				/**
+				 * Get the display settings model.
+				 *
+				 * Model returned is updated with the current customized display settings,
+				 * and an event listener is added so that changes made to the settings
+				 * will sync back into the model storing the session's customized display
+				 * settings.
+				 *
+				 * @param {Backbone.Model} model Display settings model.
+				 * @returns {Backbone.Model} Display settings model.
+				 */
+				display: function getDisplaySettingsModel( model ) {
+					var display;
+					display = wp.media.controller.Library.prototype.display.call( this, model );
+
+					display.off( 'change', this.handleDisplaySettingChange ); // Prevent duplicated event handlers.
+					display.set( customizedDisplaySettings.attributes );
+					if ( 'custom' === customizedDisplaySettings.get( 'link_type' ) ) {
+						display.linkUrl = customizedDisplaySettings.get( 'link_url' );
+					}
+					display.on( 'change', this.handleDisplaySettingChange );
+					return display;
+				}
+			} );
+
+			library = new CustomizedDisplaySettingsLibrary( {
```

Comment:
Any objections against bringing this back inline?

It would make a 23 line change a two line change (and it would save  us a variable)